ledger/acctonline.go

```diff
@@ -501,26 +470,27 @@ func (ao *onlineAccounts) postCommit(ctx context.Context, dcc *deferredCommitCon
 	ao.deltas = ao.deltas[offset:]
 	ao.deltasAccum = ao.deltasAccum[offset:]
 	ao.cachedDBRoundOnline = newBase
-	ao.expirations = ao.expirations[expirationOffset:]
-
-	// simply contatenate since both sequences are sorted
-	ao.expirations = append(ao.expirations, dcc.onlineAccountExpirations...)
-
-	maxOnlineLookback := int(ao.maxBalLookback()) + len(ao.deltas)
-	if len(ao.onlineRoundParamsData) > maxOnlineLookback {
-		ao.onlineRoundParamsData = ao.onlineRoundParamsData[len(ao.onlineRoundParamsData)-maxOnlineLookback:]
-	}
-
-	// online accounts defines deletion round as (see onlineAccountsNewRoundImpl)
-	// targetRound := basics.Round(updRound + proto.MaxBalLookback)
-	// and deletes these entries at round newBase (see ao.prepareCommit)
-	// and the original condition
-	// if ao.expirations[i].rnd <= dcc.oldBase+basics.Round(offset)
-	// becomes
-	// targetRound <= newBase
-	// ==> basics.Round(updRound + proto.MaxBalLookback) <= newBase
-	// ==> updRound <= newBase - proto.MaxBalLookback
-	ao.onlineAccountsCache.prune(newBase.SubSaturate(basics.Round(ao.maxBalLookback())))
+
+	// convert dcc.lowestRound to votersLookback relative to newBase
+	// onlineRoundParamsData contains data for newBase + 1 - maxOnlineLookback
+	minRound := (newBase + 1).SubSaturate(basics.Round(ao.maxBalLookback()))
+	if dcc.lowestRound > 0 && dcc.lowestRound < minRound {
+		minRound = dcc.lowestRound
+	}
+	// recalculate minRound back to max lookback
+	// minRound is either newBase + 1 - MaxBalLookback or less, so newBase + 1 >= minRound
+	maxLookback := newBase + 1 - minRound
+	onlineRoundParamsLookback := uint64(maxLookback) + uint64(len(ao.deltas))
+	if uint64(len(ao.onlineRoundParamsData)) > onlineRoundParamsLookback {
+		ao.onlineRoundParamsData = ao.onlineRoundParamsData[uint64(len(ao.onlineRoundParamsData))-onlineRoundParamsLookback:]
```

Comment:
since `onlineRoundParamsData` is not used in `onlineTop`, maybe it also should not track the extended history. But it introduces some disconnect from the online account history lookup that expects `onlineRoundParamsData` to mirror what is in DB. Opinions?